src/Magnum/Platform/GlfwApplication.h

```diff
@@ -0,0 +1,753 @@
+            LeftSuper = GLFW_KEY_LEFT_SUPER,    /**< Left super */
+            RightShift = GLFW_KEY_RIGHT_SHIFT,  /**< Right shift */
+            RightCtrl = GLFW_KEY_RIGHT_CONTROL, /**< Right control */
+            RightAlt = GLFW_KEY_RIGHT_ALT,      /**< Right alt */
+            RightSuper = GLFW_KEY_RIGHT_SUPER,  /**< Right super */
+            Menu = GLFW_KEY_MENU,   /**< Menu */
+        };
+
+        /** @brief Key */
+        constexpr Key key() const { return _key; }
+
+    private:
+        constexpr KeyEvent(Key key): _key(key) {}
+
+        const Key _key;
+};
+
+/**
+@brief Mouse event
+
+@see @ref MouseMoveEvent, @ref MouseScrollEvent, @ref mousePressEvent(), @ref mouseReleaseEvent()
+*/
+class GlfwApplication::MouseEvent: public GlfwApplication::InputEvent {
+    friend GlfwApplication;
+
+    public:
+        /**
+         * @brief Mouse button
+         *
+         * @see @ref button()
+         */
+        enum class Button: int {
+            Left = GLFW_MOUSE_BUTTON_LEFT,        /**< Left button */
+            Middle = GLFW_MOUSE_BUTTON_MIDDLE,    /**< Middle button */
+            Right = GLFW_MOUSE_BUTTON_RIGHT,      /**< Right button */
+            Button1 = GLFW_MOUSE_BUTTON_1,        /**< Mouse Button 1 */
+            Button2 = GLFW_MOUSE_BUTTON_2,        /**< Mouse Button 2 */
+            Button3 = GLFW_MOUSE_BUTTON_3,        /**< Mouse Button 3 */
+            Button4 = GLFW_MOUSE_BUTTON_4,        /**< Mouse Button 4 */
+            Button5 = GLFW_MOUSE_BUTTON_5,        /**< Mouse Button 5 */
+            Button6 = GLFW_MOUSE_BUTTON_6,        /**< Mouse Button 6 */
+            Button7 = GLFW_MOUSE_BUTTON_7,        /**< Mouse Button 7 */
+            Button8 = GLFW_MOUSE_BUTTON_8,        /**< Mouse Button 8 */
+        };
+
+        /** @brief Button */
+        constexpr Button button() const { return _button; }
+
+    private:
+        constexpr MouseEvent(Button button): _button(button) {}
+
+        const Button _button;
+};
+
+/**
+@brief Mouse move event
+
+@see @ref MouseEvent, @ref MouseScrollEvent, @ref mouseMoveEvent()
+*/
+class GlfwApplication::MouseMoveEvent: public GlfwApplication::InputEvent {
+    friend GlfwApplication;
+
+    public:
+
+        /** @brief Position */
+        constexpr Vector2i position() const { return _position; }
+
+    private:
+        constexpr MouseMoveEvent(const Vector2i& position): _position(position) {}
+
+        const Vector2i _position;
+};
+
+/**
+@brief Mouse scroll event
+
+@see @ref MouseEvent, @ref MouseMoveEvent, @ref mouseScrollEvent()
+*/
+class GlfwApplication::MouseScrollEvent: public GlfwApplication::InputEvent {
+    friend GlfwApplication;
+
+    public:
+
+        /** @brief Scroll offset */
+        constexpr Vector2d offset() const { return _offset; }
+
+    private:
+        constexpr MouseScrollEvent(const Vector2d& offset): _offset(offset) {}
+
+        const Vector2d _offset;
+};
```

Comment:
The thing is that the scroll `offset`is a double, but for my simple mouse just returns values in `{(0.0, 1.0), (0.0, -1.0)}). This could be easily mapped to the SDL way of handling scrolling, but could break for more sophisticated scrolling devices.

SDL [does have something similar: SDL_MouseWheelEvent](https://wiki.libsdl.org/SDL_MouseWheelEvent)